aws/resource_aws_cloudfront_realtime_log_config.go

```diff
@@ -0,0 +1,245 @@
+
+	input := &cloudfront.CreateRealtimeLogConfigInput{
+		EndPoints:    expandCloudFrontEndPoints(d.Get("endpoint").([]interface{})),
+		Fields:       expandStringSet(d.Get("fields").(*schema.Set)),
+		Name:         aws.String(d.Get("name").(string)),
+		SamplingRate: aws.Int64(int64(d.Get("sampling_rate").(int))),
+	}
+
+	log.Printf("[DEBUG] Creating CloudFront Real-time Log Config: %s", input)
+	output, err := conn.CreateRealtimeLogConfig(input)
+
+	if err != nil {
+		return fmt.Errorf("error creating CloudFront Real-time Log Config: %w", err)
+	}
+
+	d.SetId(aws.StringValue(output.RealtimeLogConfig.ARN))
+
+	return resourceAwsCloudFrontRealtimeLogConfigRead(d, meta)
+}
+
+func resourceAwsCloudFrontRealtimeLogConfigRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	logConfig, err := finder.RealtimeLogConfigByARN(conn, d.Id())
+
+	if isAWSErr(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig, "") {
+		log.Printf("[WARN] CloudFront Real-time Log Config (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading CloudFront Real-time Log Config (%s): %w", d.Id(), err)
+	}
+
+	if logConfig == nil {
+		log.Printf("[WARN] CloudFront Real-time Log Config (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("arn", logConfig.ARN)
+	if err := d.Set("endpoint", flattenCloudFrontEndPoints(logConfig.EndPoints)); err != nil {
+		return fmt.Errorf("error setting endpoint: %w", err)
+	}
+	if err := d.Set("fields", flattenStringSet(logConfig.Fields)); err != nil {
+		return fmt.Errorf("error setting fields: %w", err)
+	}
+	d.Set("name", logConfig.Name)
+	d.Set("sampling_rate", int(aws.Int64Value(logConfig.SamplingRate)))
+
+	return nil
+}
+
+func resourceAwsCloudFrontRealtimeLogConfigUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	input := &cloudfront.UpdateRealtimeLogConfigInput{
+		ARN:          aws.String(d.Id()),
+		EndPoints:    expandCloudFrontEndPoints(d.Get("endpoint").([]interface{})),
+		Fields:       expandStringSet(d.Get("fields").((*schema.Set))),
+		SamplingRate: aws.Int64(int64(d.Get("sampling_rate").(int))),
+	}
+
+	log.Printf("[DEBUG] Updating CloudFront Real-time Log Config: %s", input)
+	_, err := conn.UpdateRealtimeLogConfig(input)
+
+	if err != nil {
+		return fmt.Errorf("error updating CloudFront Real-time Log Config (%s): %s", d.Id(), err)
+	}
+
+	return resourceAwsCloudFrontRealtimeLogConfigRead(d, meta)
+}
+
+func resourceAwsCloudFrontRealtimeLogConfigDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	log.Printf("[DEBUG] Deleting CloudFront Real-time Log Config (%s)", d.Id())
+	_, err := conn.DeleteRealtimeLogConfig(&cloudfront.DeleteRealtimeLogConfigInput{
+		ARN: aws.String(d.Id()),
+	})
+
+	if isAWSErr(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig, "") {
```

Comment:
Nit:
```suggestion
	if tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig) {
```

Reply:
I'll add the `tfawserr` import and fix this (and any others).